### ISSUE TITLE
BUG: loc aligning rhs with list indexer but not with scalar

### ISSUE THREAD

**Issue by phofl** [pandas-dev/pandas#47582]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
result = pd.DataFrame({"var": [1, 2]}, index=[1, 2])

rhs = pd.DataFrame({"a": [10, 11]}, index=[2, 3])

result.loc[:, ["var"]] = rhs

This aligns the column names and hence returns

   var
1  NaN
2  NaN

while

result = pd.DataFrame({"var": [1, 2]}, index=[1, 2])

rhs = pd.DataFrame({"a": [10, 11]}, index=[2, 3])

result.loc[:, "var"] = rhs

returns 

    var
1   NaN
2  10.0
```

### Issue Description

Both cases should be consistent, I think both of them should align

cc @jbrockmendel 

### Expected Behavior

Both align

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

**Comment by simonjayhawkins** [comment]
> Both cases should be consistent, I think both of them should align

makes sense to me. We are setting values on a DataFrame (lhs) in both cases, so it should not be relevant whether the lhs indexer is a Series/DataFrame as the rhs is also a DataFrame in both cases and so we expect alignment to occur.

appears to be long standing behavior though.

**Comment by phofl** [comment]
The multi-block case already aligns for scalar indexers too, so we only have to focus on the single block case

**Comment by phofl** [comment]
Hi,

I am not sure where this is located exactly without looking myself. I would recommend one of the issues with the label good first issue if you want to get started. Indexing stuff can be complicated to fix.

**Comment by stevenhastings** [comment]
Fair enough! Thank you.

**Comment by ramtrikaavya-stack** [comment]
take
